Mastodon/Scene/Profile/ProfileViewController.swift

```diff
@@ -631,12 +631,28 @@ extension ProfileViewController: TabBarPagerDelegate {
             } else {
                 profileHeaderViewController.profileHeaderView.followsYouMaskView.frame.origin.y = .zero
             }
-            
+
+            let avatarProgress: CGFloat
+            let bottomEdge = tabBarPagerController.containerScrollView.safeAreaInsets.top + ProfileHeaderView.avatarImageViewSize.height / 2
+            if bannerContainerBottomOffset < bottomEdge {
+                let offset = bannerContainerBottomOffset - bottomEdge
+                // the progress for header move from banner bottom to header bottom (from 0 to 1)
+                avatarProgress = height != .zero ? abs(offset) / height : 0
+            } else {
+                avatarProgress = 0
+            }
+
             // setup titleView offset and fade avatar
-            profileHeaderViewController.updateHeaderScrollProgress(progress, throttle: throttle)
+            profileHeaderViewController.updateHeaderScrollProgress(avatarProgress, throttle: throttle)
             
             // setup buttonBar shadow
             profilePagingViewController.updateButtonBarShadow(progress: progress)
+
+            if avatarProgress >= throttle {
+                profileHeaderViewController.profileHeaderView.sendSubviewToBack(profileHeaderViewController.profileHeaderView.avatarImageViewBackgroundView)
+            } else {
+                profileHeaderViewController.profileHeaderView.bringSubviewToFront(profileHeaderViewController.profileHeaderView.avatarImageViewBackgroundView)
```

Comment:
This part may warrant some performance tuning as the current implementation is fairly naive in that it's constantly sending the view forward or backward. This is only really needed when avatarProgress crosses the throttle threshold.